File: sample-ktx/src/main/java/com/kpstv/sample_ktx/SQLDelightAdapters.java
package com.kpstv.sample_ktx;

import com.squareup.sqldelight.ColumnAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class SQLDelightAdapters {
    public static ColumnAdapter<List<Data>, String> SampleUserAdapter() {
        return new ColumnAdapter<List<Data>, String>() {
            @NotNull
            @Override
            public List<Data> decode(String s) {
                return null;
            }

            @Override
            public String encode(@NotNull List<Data> data) {
                return null;
            }
        };
    }
}

File: sample-ktx/src/main/java/com/kpstv/sample_ktx/SQLDelightAdapters.kt
package com.kpstv.sample_ktx

import com.kpstv.bindings.*
import com.squareup.sqldelight.ColumnAdapter

@AutoGenerateSQLDelightAdapters(using = ConverterType.KOTLIN_SERIALIZATION)
interface SQLDelightAdapters {
    fun tagsConverter(): ColumnAdapter<List<String>, String>
}